components/sp-solutions/org.wso2.analytics.solutions.sp.message.tracer/widgets/OpenTracingVisTimeline/src/OpenTracingVisTimeline.jsx

```diff
@@ -242,84 +297,136 @@ class OpenTracingVisTimeline extends Widget {
                 },
                 groupTemplate: function (group) {
                     let container = document.createElement('div');
-                    let label = document.createElement('span');
-                    label.innerHTML = group.content + ' ';
-                    container.insertAdjacentElement('afterBegin', label);
+                    const kindData = kindsData[group.kind];
+                    ReactDOM.render((
+                        <div>
+                            <div style={{minWidth: '150px'}} className="spanLabelContainer">
+                                <span class='serviceName'>{group.content}</span>
+                                <span class='operationName'>{group.operationName}</span>
+                            </div>
+                            {(kindData ? <div className="kindBadge" style={{backgroundColor: kindData.color}}>
```

Comment:
Instead of this, you can write it as follows.

```jsx
{
    kindData && <div className="kindBadge" style={{backgroundColor: kindData.color}}>{kindData.name}</div>
}